packages/gatsby/src/utils/merge-gatsby-config.js

```diff
@@ -0,0 +1,16 @@
+const _ = require("lodash")
+/**
+ * Defines how a theme object is merged with the user's config
+ */
+module.exports = (a, b) =>
+  _.uniq(Object.keys(a).concat(Object.keys(b))).reduce((acc, key) => {
+    const mergeFn = mergeAlgo[key]
+    acc[key] = mergeFn ? mergeFn(a[key], b[key]) : b[key] || a[key]
```

Comment:
A comment clarifying this logic might be helpful for future maintainers. Something to the effect of, "If a merge function is not defined, the config option will be overridden."

I say this because it took me a bit to think through this logic and understand how e.g. `proxy` overrides but plugins merge.

Reply:
I'm open to rewriting this too. Do you think putting the fallback into the list of algorithms would be more approachable? Could combine it with spreading to make it a bit less "compressed".

```js
.reduce((acc, key) => {
    const mergeFn = mergeAlgo[key] || mergeAlgo.fallback
    return {
      ...acc,
      [key]: mergeFn(a[key], b[key])
    }
})
...

{
  fallback: (a,b) => b || a.
  ...
}
```

Reply:
That could help for sure!

Perhaps naming would help in this case, too? E.g. `mergedConfig` vs. `acc`, `overrideOption` vs. `fallback`, `mergeConfigOptions` vs. `mergeFn`?

We're now crossing over into Personal Taste™, but my experience has been that Future Jason tends to curse Past Jason far less for verbose variable names. 😅